airbyte-webapp/src/views/Connection/ConnectionForm/components/SchemaChangesBackdrop.test.tsx

```diff
@@ -0,0 +1,86 @@
+import { render } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { mockConnection, TestWrapper } from "test-utils/testutils";
+
+import { SchemaChange } from "core/request/AirbyteClient";
+import { FeatureItem } from "hooks/services/Feature";
+const mockUseConnectionEditService = jest.fn();
+
+jest.doMock("hooks/services/ConnectionEdit/ConnectionEditService", () => ({
+  useConnectionEditService: mockUseConnectionEditService,
+}));
+
+const TestWrapperWithAutoDetectSchema: React.FC<React.PropsWithChildren<Record<string, unknown>>> = ({ children }) => (
+  <TestWrapper features={[FeatureItem.AllowAutoDetectSchemaChanges]}>{children}</TestWrapper>
+);
+
+const buttonSpy = jest.fn();
+
+const renderComponent = () =>
+  render(
+    <SchemaChangeBackdrop>
+      <button data-testid="bg-button" onClick={() => buttonSpy}>
+        don't click
+      </button>
+    </SchemaChangeBackdrop>,
+    { wrapper: TestWrapperWithAutoDetectSchema }
+  );
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const { SchemaChangeBackdrop } = require("./SchemaChangeBackdrop");
+
+describe("SchemaChangesBackdrop", () => {
+  it("renders with breaking changes and prevents background interaction", () => {
+    mockUseConnectionEditService.mockReturnValue({
+      connection: { mockConnection, schemaChange: SchemaChange.breaking },
+      schemaHasBeenRefreshed: false,
+      schemaRefreshing: false,
+    });
+
+    const { getByTestId } = renderComponent();
+
+    expect(getByTestId("schemaChangesBackdrop")).toMatchSnapshot();
+    userEvent.click(getByTestId("bg-button"));
+    expect(buttonSpy).not.toHaveBeenCalled();
```

Comment:
testing that background elements aren't clickable... though in reality, the implementation ends up dependent on the z-index of elements within the connection form.  i didn't feel that implementing a whole integration test to include the actual form elements in this was a good use of time for this PR.